workflow/executor/executor.go

```diff
@@ -315,15 +332,61 @@ func (we *WorkflowExecutor) InitDriver(art wfv1.Artifact) (artifact.ArtifactDriv
 
 // getPod is a wrapper around the pod interface to get the current pod from kube API server
 func (we *WorkflowExecutor) getPod() (*apiv1.Pod, error) {
-	// TODO(jessesuen): harden this as part of issue #653
-	podIf := we.ClientSet.CoreV1().Pods(we.Namespace)
-	pod, err := podIf.Get(we.PodName, metav1.GetOptions{})
+	podsIf := we.ClientSet.CoreV1().Pods(we.Namespace)
+	var pod *apiv1.Pod
+	var err error
+	_ = wait.ExponentialBackoff(DefaultRetry, func() (bool, error) {
+		pod, err = podsIf.Get(we.PodName, metav1.GetOptions{})
```

Comment:
The underlying client-go request seems to have some built in retry for GET. You may want to check to see if it already meets your retry requirement.

Reply:
Looks like this mechanism does not suit our needs. The issue we are trying to solve is the executor retrying getting pod/secrets from API server to address issues like this:

```
   ├-⚠ hybridize(23)              hm-92j79-3168642571  failed to load artifacts: Get https://10.11.240.1:443/api/v1/namespaces/default/secrets/gcs-credentials: dial tcp 10.11.240.1:443: getsockopt: connection refused
```